hack/verify-shellcheck.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck source=/dev/null
+ROOT_PATH=$(git rev-parse --show-toplevel)
+
+# create a temporary directory
+TMP_DIR=$(mktemp -d)
+
+# cleanup on exit
+cleanup() {
+  echo "Cleaning up..."
+  rm -rf "${TMP_DIR}"
+}
+trap cleanup EXIT
+
+# check for shellcheck installation
+SHELLCHECK_VERSION=0.6.0
+HAVE_SHELLCHECK=false
+if which shellcheck &>/dev/null; then
+  detected_version="$(shellcheck --version | grep 'version: .*')"
+  if [[ "${detected_version}" = "version: ${SHELLCHECK_VERSION}" ]]; then
+    HAVE_SHELLCHECK=true
+  fi
+fi
+if ! ${HAVE_SHELLCHECK}; then
+  # install shellcheck (Linux-x64 only!)
+  if [[ "$OSTYPE" != "darwin"* ]]; then
+    cd "${TMP_DIR}" || exit
+    VERSION="shellcheck-v0.6.0"
+    DOWNLOAD_FILE="${VERSION}.linux.x86_64.tar.xz"
+    wget https://storage.googleapis.com/shellcheck/"${DOWNLOAD_FILE}"
+    tar xf "${DOWNLOAD_FILE}"
+    cd "${VERSION}" || exit
+  fi
+fi
+
+echo "Running shellcheck..."
+cd "${ROOT_PATH}" || exit
+OUT="${TMP_DIR}/out.log"
+FILES=$(grep -rl '#!/bin/bash')
```

Comment:
`grep: warning: recursive search of stdin`
```suggestion
FILES=$(grep -rl '#!/bin/bash' . 2>/dev/null)
```

Reply:
Interesting! I guess MacOS gives a warning with recursive grep. I implemented this in a new push